packages/mco/components/modals/app-manage-policies/unassign-policy-view.tsx

```diff
@@ -0,0 +1,193 @@
+  UnAssignPolicyViewState,
+} from './utils/reducer';
+import { DRPlacementControlType } from './utils/types';
+
+export const GeneratefooterButtons: React.FC<GeneratefooterButtonsProps> = ({
+  drpcs,
+  isAllSelected,
+  modalActionContext,
+  setModalContext,
+  setModalActionContext,
+  setMessage,
+  dispatch,
+}) => {
+  const { t } = useCustomTranslation();
+  const [isButtonLoading, setButtonLoading] = React.useState(false);
+
+  const onBack = () => setModalContext(ModalViewContext.POLICY_LIST_VIEW);
+  const onUnassign = () => {
+    setModalActionContext(ModalActionContext.UN_ASSIGNING_POLICY);
+    setMessage({
+      title: t(
+        'Unassign policy will no longer DR protect your application, other applications sharing the same placement will also be affected.'
+      ),
+      variant: AlertVariant.warning,
+    });
+  };
+  const onCancel = () => setModalActionContext(undefined);
+
+  const unAssignPolicy = () => {
+    setButtonLoading(true);
+    // unassign DRPolicy
+    const promises = unAssignPromises(drpcs);
+    Promise.all(promises)
+      .then(() => {
+        setButtonLoading(false);
+        // Display success message
+        setModalActionContext(
+          ModalActionContext.UNASSIGN_POLICY_SUCCEEDED,
+          ModalViewContext.POLICY_LIST_VIEW
+        );
+        setMessage(
+          {
+            title: t('Policy unassigned for the application.'),
+            variant: AlertVariant.success,
+          },
+          ModalViewContext.POLICY_LIST_VIEW
+        );
+        // Switch to list policy view
+        setModalContext(ModalViewContext.POLICY_LIST_VIEW);
+        dispatch({
+          type: ManagePolicyStateType.SET_SELECTED_POLICY,
+          context: ModalViewContext.UNASSIGN_POLICY_VIEW,
+          payload: undefined,
+        });
+      })
+      .catch((error) => {
+        setButtonLoading(false);
+        setMessage({
+          title: t('Unable to unassign policy for the application.'),
+          description: getErrorMessage(error),
+          variant: AlertVariant.danger,
+        });
+      });
+  };
+
+  return modalActionContext === ModalActionContext.UN_ASSIGNING_POLICY ? (
+    <>
+      <Button
+        id="modal-cancel-action"
+        variant={ButtonVariant.secondary}
+        onClick={onCancel}
+      >
+        {t('Cancel')}
+      </Button>
+      <Button
+        id="modal-confirm-action"
+        variant={ButtonVariant.primary}
+        onClick={unAssignPolicy}
+        isLoading={isButtonLoading}
+      >
+        {t('Confirm')}
+      </Button>
+    </>
+  ) : (
+    <>
+      <Button
+        id="modal-back-action"
+        variant={ButtonVariant.secondary}
+        onClick={onBack}
+      >
+        {t('Back')}
+      </Button>
+      <Button
+        id="modal-unassign-action"
+        variant={ButtonVariant.primary}
+        onClick={onUnassign}
+      >
+        {isAllSelected ? t('Unassign for all') : t('Unassign')}
+      </Button>
+    </>
+  );
+};
```

Comment:
nits (but feel free to ignore this comment if want to):
```suggestion
const isContextUnAssign = modalActionContext === ModalActionContext.UN_ASSIGNING_POLICY;
  return  ? (
    <>
      <Button
        id="modal-cancel-action"
        variant={ButtonVariant.secondary}
        onClick={isContextUnAssign ? onCancel : onBack}
      >
        {isContextUnAssign ? t('Cancel') : t('Back')}
      </Button>
      <Button
        id="modal-confirm-action"
        variant={ButtonVariant.primary}
        onClick={isContextUnAssign ? unAssignPolicy: onUnassign}
        isLoading={isButtonLoading}
      >
        {isContextUnAssign ? ('Confirm') : isAllSelected ? t('Unassign for all') : t('Unassign')}
      </Button>
    </>
  )
};
```